src/components/ChatEntry.js

```diff
@@ -1,22 +1,42 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp'
+
 
 const ChatEntry = (props) => {
+
+    const onHeartButtonClick = () => {
+        const updatedHeart = {
+            id: props.id,
+            sender: props.sender,
+            body: props.body,
+            timeStamp: props.timeStamp,
+            liked: !props.liked
+        };
+
+        props.onUpdateChat(updatedHeart);
+    };
+
+  const heartColor = props.liked ? '❤️':'🤍';
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+      <h2 className="entry-name">{props.sender}</h2>
+      <section className="entry-bubble">{props.body}
+        <p className="entry-time"><TimeStamp time={props.timeStamp}> </TimeStamp></p>
+        <button className="like" onClick={onHeartButtonClick}>{heartColor}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  onUpdateChat: PropTypes.func
```

Comment:
```suggestion
  onUpdateChat: PropTypes.func.isRequired
```
I'd recommend making this required here too since its essential for the like functionality